satnogsclient/observer/orbital.py

```diff
@@ -1,56 +1,106 @@
-    """
-    # observer object
-    if all(map(lambda x: x in observer_dict, ['lat', 'lon', 'elev'])):
-        observer = ephem.Observer()
-        observer.lon = str(observer_dict['lon'])
-        observer.lat = str(observer_dict['lat'])
-        observer.elevation = float(observer_dict['elev'])
-    else:
-        return {'ok': False}
-
-    # satellite object
-    if all(map(lambda x: x in satellite_dict, ['tle0', 'tle1', 'tle2'])):
-        tle0 = str(satellite_dict['tle0'])
-        tle1 = str(satellite_dict['tle1'])
-        tle2 = str(satellite_dict['tle2'])
-        try:
-            satellite = ephem.readtle(tle0, tle1, tle2)
-        except ValueError:
-            print(('error:', 'ephem object', 'tle values', sys.exc_info()[0]))
-            return {'ok': False}
-        except:
-            print(('error:', 'ephem object', sys.exc_info()[0]))
-            return {'ok': False}
-    else:
-        return {'ok': False}
-
-    # time of observation
-    if not timestamp:
-        timestamp = datetime.now(pytz.utc)
-
-    # observation calculation
-    observer.date = timestamp.strftime('%Y-%m-%d %H:%M:%S.%f')
-    satellite.compute(observer)
-    return {'alt': satellite.alt, 'az': satellite.az,
-            'rng': satellite.range, 'rng_vlct': satellite.range_velocity,
-            'ok': True}
+        # satellite object
+        if satellite_dict:
+            ret = self._set_satellite_dict(satellite_dict)
+            if not ret:
+                return False
+
+        # no problems
+        return True
+
+    def _set_observer(self, observer_dict):
+        """
+        Uses the dictionary provided to set observer attributes used in pyephem calculations.
+        """
+        if observer_dict:
+            if all([x in observer_dict for x in ['lat', 'lon', 'elev']]):
+                self.observer = ephem.Observer()
+                self.observer.lon = str(observer_dict['lon'])
+                self.observer.lat = str(observer_dict['lat'])
+                self.observer.elevation = float(observer_dict['elev'])
+                return True
+        return False
+
+    def _set_satellite(self, satellite_dict):
+        """
+        Uses the dictionary provided to set satellite trajectory attributes used in pyephem calculations.
+        """
+        if satellite_dict:
+            if all([x in satellite_dict for x in ['tle0', 'tle1', 'tle2']]):
+                tle0 = str(satellite_dict['tle0'])
+                tle1 = str(satellite_dict['tle1'])
+                tle2 = str(satellite_dict['tle2'])
+                try:
+                    self.satellite = ephem.readtle(tle0, tle1, tle2)
+                    return True
+                except ValueError:
+                    #return {'ok': False,, 'error': 'ephem object - tle values problematic: {}'.format(sys.exc_info()[0])}
+                    return False
+                except:
+                    #return {'ok': False,, 'error': 'ephem object: {}'.format(sys.exc_info()[0])}
```

Comment:
Buy returning the same thing when catching a superset of the previous exception you are practically doing the (more complicated) equivalent of 

``
try:
   something
except:
   return False
```